### PR TITLE
Add controller for MksPipelineRun

### LINKED ISSUE #12: create mks-pipelinerun controller

### PR DESCRIPTION
**Pull request by UtkarshChaurasia**
### Add controller for MksPipelineRun

- Listens for MksPipelineRun resource being created, updated, or deleted.
- Adds corresponding TaskPipeline using Tekton API.

### How to test

- Build and run the executable using steps in README.md.
- Apply the CRDs and examples in the config folder.
- Add a new MksPipelineRun resource and see for logs.

### Related issues

Fixes #12

### CODE REVIEW

**Comment by UtkarshChaurasia** [comment]
Hello @piyush-garg, please review.

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,4 @@ vendor/
 
 .vscode
 .idea
+
```
Comment (avinal) [inline .gitignore]:
```suggestion
```
Remove newline changes (unnecessary)

File: main.go [hunk 2]
```diff
@@ -1,9 +1,90 @@
...
+	fmt.Println(list)
+
+	gt, err := exampleClient.MkscontrollerV1alpha1().MksPipelineRuns("default").List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		klog.Fatalf("Error listing all databases: %v", err)
+	}
+	for _, name := range gt.Items {
+		fmt.Println(name)
+
+	}*/
```
Comment (avinal) [inline main.go]:
```suggestion
```

File: main.go [hunk 3]
```diff
@@ -1,9 +1,90 @@
...
+	"flag"
 	"fmt"
+	"time"
+
+	//"github.com/MiniTeks/mks-server/pkg/apis/mkscontroller/v1alpha1"
+	examplecomclientset "github.com/MiniTeks/mks-server/pkg/client/clientset/versioned"
+	informers "github.com/MiniTeks/mks-server/pkg/client/informers/externalversions"
+	con "github.com/MiniTeks/mks-server/pkg/controllers/mkspipelinerun"
+
+	//v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (avinal) [inline main.go]:
```suggestion
```

File: main.go [hunk 4]
```diff
@@ -1,9 +1,90 @@
 package main
 
 import (
+	//"context"
+	"flag"
 	"fmt"
+	"time"
+
+	//"github.com/MiniTeks/mks-server/pkg/apis/mkscontroller/v1alpha1"
```
Comment (avinal) [inline main.go]:
```suggestion
```

File: main.go [hunk 5]
```diff
@@ -1,9 +1,90 @@
 package main
 
 import (
+	//"context"
```
Comment (avinal) [inline main.go]:
```suggestion
```

**Comment by avinal** [comment]
Please remove commented code and sqash all commits to at most 2 commits.

**Comment by UtkarshChaurasia** [comment]
> Please remove commented code and sqash all commits to at most 2 commits.

Thank you. Did the suggested changes

**Review by ramessesii2 — APPROVED** [review]

**Pull request merged**